mne/decoding/transformer.py

```diff
@@ -245,6 +246,111 @@ def inverse_transform(self, X, y=None):
         return X.reshape(-1, self.n_channels, self.n_times)
 
 
+class Vectorizer(TransformerMixin):
+    """Transforms n-dimensional array into 2D array of n_samples by n_features.
+
+    This class reshapes an n-dimensional array into an n_samples * n_features
+    array, usable by the estimators and transformers of scikit-learn.
+
+    Examples
+    --------
+    clf = make_pipeline(SpatialFilter(), XdawnTransformer(), Vectorizer(),
+                        LogisticRegression())
+
+    Attributes
+    ----------
+    ``features_shape_`` : tuple
+         Stores the original shape of data.
+    """
+
+    def fit(self, X, y=None):
+        """Stores the shape of the features of X.
+
+        Parameters
+        ----------
+        X : array-like
```

Comment:
@agramfort Since the array could be of multiple shapes, how should it be written? This is similar to how scikit-learn [doc](https://github.com/scikit-learn/scikit-learn/blob/51a765a/sklearn/cross_validation.py#L1362) is written